### ISSUE TITLE
Using JoinIdenticalVertices multiplies the final file size

### ISSUE THREAD

**Issue by wlinna** [assimp/assimp#3641]
**Describe the bug**
Using `JoinIdenticalVertices` flag multiplies the final file size. For example, a 22 MiB GLB turns into an 88 MiB monster if I use the flags `-jiv`, but after removing `-jiv` the final file size is 33 MiB. 

Through the debug logs I discovered that if I use `-jiv`, the `MakeVerboseFormat` is triggered in the export phase, but the supposed `JoinVertices` is NOT -- In fact, `JoinVertices` is run BEFORE `Exporter::Export`. It seems like `must_join_again = true;` isn't called at all... maybe because `JoinIdenticalVertices` has already been flagged as a completed step.

**To Reproduce**
1. `.\assimp.exe export .\<your_glb> .\output_jiv.glb -jiv -v -l`
2. `.\assimp.exe export .\<your_glb> .\output_no_jiv.glb -v -l`
3. Observe the file sizes and logs

**Expected behavior**
`MakeVerboseFormat` shouldn't be called if there are no steps to run anymore. Or if it's called, at least `JoinVertices` should be called afterwards. The file size shouldn't explode.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Assimp version: this happens both on the tip of the master branch and with my old version of Assimp

**Additional context**
Actually, I use many more post-processing flags, and it seems like all of them are called before the `Exporter::Export`. `MakeVerboseFormat` gets only ever called afterwards and the only process called afterwards is  `SortByPType`

**Comment by kimkulling** [comment]
Indeed, this destroys the main propose for this post-processing step.- Thanks a lot for your repport.

**Comment by tellypresence** [comment]
@wlinna Is this still a problem?

**Comment by wlinna** [comment]
@tellypresence  I just tested, this is still a problem

**Comment by tellypresence** [comment]
@wlinna Thank you for following up, reopening

**Comment by tellypresence** [comment]
glTF export issues tracked via epic #6147

**Comment by wlinna** [comment]
I see that you've included this in glTF export related bugs, but the way I remember it, this issue is not directly related to glTF. I think it concerns every format. It's been some time since I checked the code so I'm not too sure
